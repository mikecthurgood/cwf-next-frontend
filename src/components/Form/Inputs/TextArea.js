import React from 'react';
import Styled from 'styled-components';

const TextArea = props => (
  <InputContainer>
    {props.label && <label htmlFor={props.id}>{props.label}</label>}
    <textarea
      className={[
        !props.valid ? 'invalid' : 'valid',
        props.touched ? 'touched' : 'untouched'
      ].join(' ')}
      id={props.id}
      rows={props.rows}
      required={props.required}
      value={props.value}
      onChange={e => props.onChange(props.id, e.target.value)}
      onBlur={props.onBlur}
      />
  </InputContainer>
);

export default TextArea;

const InputContainer = Styled.div`
    margin: 1rem 0;
    width: 100%;
    overflow: hidden;

  label {
    display: block;
    text-transform: uppercase;
    margin-bottom: 0.25rem;
    color: #282c34;
  }

  textarea {
    display: block;
    font: inherit;
    padding: 0.25rem 0.5rem;
    width: 100%;
    border-radius: 3px;
    border: 1px solid #ccc;
  }

  .touched.invalid {
      border-color: red;
      background: #ffc2c2;
  }

  textarea:focus {
    outline: none;
    border-color: #3b0062;
    color: #3b0062;
  }
`
