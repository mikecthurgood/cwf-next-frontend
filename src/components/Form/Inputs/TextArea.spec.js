import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import TextArea from './TextArea';

describe('Form Text Area', () => {
    afterEach(cleanup);
    it('render the component', () => {
            const props = {
                valid: false,
                touched: false,
                loginError: false,
            };
    
        const { container } = render(
            <TextArea {...props} />
        );

        expect(container).toMatchSnapshot();
    });

    it('render the component', () => {
        const props = {
            valid: true,
            touched: true,
            loginError: true,
            label: 'test label'
        };

        const { container } = render(
            <TextArea {...props} />
        );

        expect(container).toMatchSnapshot();
    });

    it('should render the component', () => {
        const props = {
            valid: true,
            touched: true,
            loginError: true,
            label: 'test label',
            onChange: jest.fn(),
        };
        const { getByTestId } = render(
            <TextArea {...props} />
        );

        const textArea = getByTestId('form-text-area');
        fireEvent.change(textArea, { target: { value: 'Good Day' } });
        expect(props.onChange).toBeCalled();
        expect(textArea.value).toBe('Good Day');
        
    })
});
