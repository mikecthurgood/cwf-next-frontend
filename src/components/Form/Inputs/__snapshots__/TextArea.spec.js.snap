// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Text Area render the component 1`] = `
<div>
  <div
    class="TextArea__InputContainer-zzvw0m-0 epbDDn"
  >
    <textarea
      class="invalid untouched"
      data-testid="form-text-area"
    />
  </div>
</div>
`;

exports[`Form Text Area render the component 2`] = `
<div>
  <div
    class="TextArea__InputContainer-zzvw0m-0 epbDDn"
  >
    <label>
      test label
    </label>
    <textarea
      class="valid touched"
      data-testid="form-text-area"
    />
  </div>
</div>
`;
